src/main.rs

```diff
@@ -1438,6 +1472,24 @@ mod cli {
         pub german_naive: bool,
     }
 
+    /// The query as read in from the CLI. This could either be a literal query or
+    /// a path to a query file.
+    #[derive(Clone, Debug)]
+    struct QuerySourceOrPath(String);
```

Comment:
Just thinking out loud here... can something like

```rust
enum QuerySourceOrPath {
    Unknown(String),
    Query(RawQuery), // or `TSQuery`, or...
    Path(PathBuf),
}
```

make sense? (no need to bikeshed this though, it's fine as is)

Reply:
I don't think it's possible to know at this point what the contents of the `String` given to us from `clap` is.
We only know that it's a file path when we try open it later on.

Of course if we decided to use a prefix symbol to signal the intent of the `String` contents, then we could parse the `String` into this type of enum.